Docs: Tweak syntax highlighting

shati-patel opened the pull request:
The syntax highlighting had gone awry in some places, e.g. interpreting non-code as QL, or highlighting QL as Python.

<details><summary>Expand for examples</summary>

In the "[Expressions](https://codeql.github.com/docs/ql-language-reference/expressions)" article:

![image](https://user-images.githubusercontent.com/42641846/102987547-67e54400-450a-11eb-8c75-243f944f5a8d.png)

![image](https://user-images.githubusercontent.com/42641846/102987537-63b92680-450a-11eb-8573-29b4277b2fb6.png)
</details> 

### Changes in this PR

I've explicitly set the default highlighting language to "none" (so it no longer defaults to Python), and fixed the indentation and highlighting language where necessary.

I ran the sphinx build locally with the `-W` flag to see any immediate warnings, and then manually checked the previews for any other bugs.

shati-patel commented on docs/codeql/codeql-language-guides/codeql-library-for-csharp.rst:
```diff
@@ -108,8 +108,8 @@ Count the number of lines of code, excluding the directory ``external``:
    select sum(SourceFile f |
-     not exists(Folder external | external.getShortName() = "external" |
-                external.getAFolder*().getAFile() = f) |
+     not exists(Folder ext | ext.getShortName() = "external" |
```
Slightly unrelated change, but the QL highlighter was highlighting the variable "external" as a keyword, so this was the easiest fix.

![image](https://user-images.githubusercontent.com/42641846/102998629-71c57200-451f-11eb-8e60-29722ff6469d.png)

shati-patel commented on docs/codeql/conf.py:
```diff
@@ -39,6 +39,10 @@
 # The name of the Pygments (syntax highlighting) style to use.
 pygments_style = 'sphinx'
 
+# The default language for syntax highlighting, defaults to Python if omitted.
+
+highlight_language = "none"
```
This will stop Sphinx from highlighting unlabeled code samples as Python!

adityasharad replied:
Update the comment above it!

shati-patel commented on docs/codeql/ql-language-reference/ql-language-specification.rst:
```diff
@@ -7,10 +7,6 @@ QL language specification
 
 This is a formal specification for the QL language. It provides a comprehensive reference for terminology, syntax, and other technical details about QL.
 
-.. This ``highlight`` directive prevents code blocks in this file being highlighted as QL (the default language for this Sphinx project). 
-
-.. highlight:: none
-
```
The [global default](https://github.com/github/codeql/pull/4874/files#r547949703) language for this Sphinx project is now "none", so there's no need to repeat this here.

hubwriter commented on docs/codeql/ql-language-reference/expressions.rst:
```diff
@@ -394,48 +406,54 @@ aggregation in a simpler form:
        count(int i | i = "hello".indexOf("l") | i)
        count("hello".indexOf("l"))
 
 #. If there only one aggregation variable, you can omit the ``<expression>`` part instead.
```
```suggestion
#. If there is only one aggregation variable, you can omit the ``<expression>`` part instead.
```

shati-patel replied:
Good spot, thank you!

hubwriter commented:
Shati - This looks good. I've done a pretty thorough check, comparing the before & after versions of each pages. I think this is good to merge. I just have a few observations.

On the whole I think the pages are rendered more nicely in the version on docteam. I'm not sure why the CSS should be different. There are a few exceptions however: 

1) The red background colour of the Important boxes here: https://codeql.github.com/docs/ql-language-reference/evaluation-of-ql-programs/#binding 
   and here: https://codeql.github.com/docs/ql-language-reference/annotations/#library
has become grey here: http://docteam.internal.semmle.com/shati/codeql/ql-language-reference/evaluation-of-ql-programs.html#binding
   and here: http://docteam.internal.semmle.com/shati/codeql/ql-language-reference/annotations.html#library

2) `docs/codeql/writing-codeql-queries/about-codeql-queries.rst` 
The 2 bulleted lines in the note box here: https://codeql.github.com/docs/writing-codeql-queries/about-codeql-queries/#query-metadata
have become bold here: http://docteam.internal.semmle.com/shati/codeql/writing-codeql-queries/about-codeql-queries.html#query-metadata 
I don't see anything in the RST to indicate they should be in bold.

3) `docs/codeql/writing-codeql-queries/cross-the-river.rst` 
The show/hide arrowheads here: https://codeql.github.com/docs/writing-codeql-queries/cross-the-river/#walkthrough
have dropped onto a new line here: http://docteam.internal.semmle.com/shati/codeql/writing-codeql-queries/cross-the-river.html#walkthrough

shati-patel commented:
Thank you so much for the thorough check, @hubwriter! 

> On the whole I think the pages are rendered more nicely in the version on docteam (better line spacing etc.). I'm not sure why the CSS should be different. 

Hmm, I'm not sure why that's the case 🤔  It's possible that my local version of Sphinx is different to the one used to generate codeql.github.com. I'll investigate!

> 1. The red background colour of the Important boxes has become grey ...

This is probably something to do with different Sphinx versions too... I've changed all of the "important" boxes to "pull quotes" in this PR: https://github.com/github/codeql/pull/4878, which should give us more consistent styling throughout!

> 2. `docs/codeql/writing-codeql-queries/about-codeql-queries.rst`
>    The 2 bulleted lines in the note box here: https://codeql.github.com/docs/writing-codeql-queries/about-codeql-queries/#query-metadata
>    have become bold here: http://docteam.internal.semmle.com/shati/codeql/writing-codeql-queries/about-codeql-queries.html#query-metadata
>    I don't see anything in the RST to indicate they should be in bold.

Ah yes, I fixed this on the live site, but the corresponding change in this repo hasn't been merged yet. Should be fixed once https://github.com/github/codeql/pull/4873 is merged!

> 3. `docs/codeql/writing-codeql-queries/cross-the-river.rst`
>    The show/hide arrowheads here: https://codeql.github.com/docs/writing-codeql-queries/cross-the-river/#walkthrough
>    have dropped onto a new line here: http://docteam.internal.semmle.com/shati/codeql/writing-codeql-queries/cross-the-river.html#walkthrough
> 
> Maybe these things will resolve themselves when published to codeql.github.com?

I have no idea why this happens on my local preview 😅  
But it looks fine once published to codeql.github.com, so I'll keep an eye on it at the next publication and see what happens 🤞🏽
